### PR TITLE
Add AssertionManager and OutputAssertionManager

### PR DESCRIPTION
**Pull request by simonjjones**
## Summary
Following on from discussion in #673 this contributes further changes intended to improve the readability, maintainability and user experience of working with our acceptance test suite.

This change creates an AssertionManager, which provides a slightly different abstraction for common types of assertion, an OutputAssertionManager which provides a way to abstract the actual content of messages away from tests checking whether the messages are printed in various scenarios

Hopefully there are no changes to the logic of what is being tested, just rearrangement for clarity.

## Documentation
- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

Contributes to #606

### CODE REVIEW

File: acceptance/assertions/output.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+func (o OutputAssertionManager) ReportsConnectedToInternet() {
+	o.testObject.Helper()
+
+	o.assert.Contains(o.output, "RESULT: Connected to the internet")
+}
+
+func (o OutputAssertionManager) ReportsDisconnectedFromInternet() {
+	o.testObject.Helper()
+
+	o.assert.Contains(o.output, "RESULT: Disconnected from the internet")
```
Comment (jromero) [inline acceptance/assertions/output.go:179]: It feels like this "manager" is a little overloaded. I can see the value in common pack specific output given that is what the project repo is for. 

On the other hand, abstracting other output that is more specific to a particular test makes it just an abstraction leading to a burden with little to no value if we were to expect the pattern to be followed.

File: acceptance/assertions/output.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+}
+
+func (o OutputAssertionManager) ReportsExporterReusingUnchangedLayer(layer string) {
+	o.testObject.Helper()
+	o.testObject.Log("exporter reuses unchanged layers")
+
+	o.assert.Matches(o.output, regexp.MustCompile(fmt.Sprintf(`(?i)Reusing layer '%s'`, layer)))
+}
+
+func (o OutputAssertionManager) ReportsCacheReuse(layer string) {
```
Comment (jromero) [inline acceptance/assertions/output.go:78]: Related to the `OutputAssertionManager` being "overloaded".

Another potential improvement here may be to create a separation for the parties producing the output. For instance this output is produced by the lifecycle and not pack so something like `LifecycleOutputAssertionManager` would further encapsulate and clarify responsibilities.

File: acceptance/acceptance_test.go [hunk 3]
```diff
@@ -55,16 +55,18 @@ func TestAcceptance(t *testing.T) {
 	h.RequireDocker(t)
 	rand.Seed(time.Now().UTC().UnixNano())
 
+	assert := h.NewAssertionManager(t)
+
 	dockerCli, err = client.NewClientWithOpts(client.FromEnv, client.WithVersion("1.38"))
-	h.AssertNil(t, err)
+	assert.Nil(err)
```
Comment (jromero) [inline acceptance/acceptance_test.go:61]: Visually this new `assert.X` style looks like a 🏆 .

Comment (simonjjones) [reply acceptance/acceptance_test.go:61]: It feels nice to work with, I've put it in `testhelpers` too, so immediately usable in all our test suites.

File: acceptance/assertions/output.go (hunk 1 above)
Comment (simonjjones) [reply acceptance/assertions/output.go:179]: Given the `LifecycleOutputAssertionManager` idea floated in a related comment, do you think something like a `TestAppOutputAssertionManager` be a worthwhile abstraction, that would ensure the separation of concerns?

Comment (simonjjones) [reply acceptance/assertions/output.go:140]: Or more accurately `TestBuildpackOutputAssertionManager`... becoming somewhat of a mouthful

Comment (jromero) [reply acceptance/assertions/output.go]: Hahaha. Yeah, `NewTestBuildpackOutputAssertionManager` is so Java-ey. I think the use of packages may help.

Naming concerns aside, it's still debatable. Without working in it first hand with these set of change I can't say that I have too strong of an opinion. I'm just wondering whether it's reasonable to expect the creation of a new method solely to call another singleline when you could have just called `assert.Contains(...)`. Specially when that method is only used once.

Either way, looking at the latest changes. I'm okay with merging this in and iterating on it if need be.

Comment (simonjjones) [reply acceptance/assertions/output.go]: Perfect

**Pull request merged**